training/architecture/DATA_TERMS.md

```diff
@@ -0,0 +1,57 @@
+Common Data Terms
+===========================
+
+This training document breaks down common terms we use to describe our data, and the relationships between different data types.
+
+## Pre-reading
+None.
+
+## Materials
+
+### What do we publish? Datasets, editions and versions
+
+**Datasets** are collections of data, published on a repeating basis. As an example, a dataset may be the Annual Count of Welsh Pets.
```

Comment:
Just a double check for my own learning rather than a comment for this PR - Are there any cases where a dataset isn't repeating and is a one-off publication? e.g. Number of movies Chuck Norris has appeared in per year - couldn't think of a potential relevant example for ONS data ><.